requirements.txt

```diff
@@ -4,3 +4,5 @@ qiskit-terra>=0.20.1
 qiskit-ibmq-provider>=0.16.0
 matplotlib>=3.4
 uncertainties
+lmfit
+asteval
```

Comment:
Is this needed? isnt it already a req for lmfit?

Reply:
You are correct.

Reply:
asteval is also explicitly imported in `eval_with_uncertainties` so it makes sense to call it out.

Reply:
Yes, this is also true. What is the advantage to explicitly write the dependency which will be installed by other packages?

Reply:
In general, it is best to list everything that is directly used as a dependency. That way if a dependency is removed or drops a subdependency in the future it does not take away a required dependency that was only included implicitly. Likely we would not want asteval is we dropped lmfit or it dropped asteval, so it's a pretty minor point, but I don't see the harm in listing the dependency on asteval.

Reply:
Sounds reasonable to me. Any thoughts? @chriseclectic

Reply:
I agree with Will, if you import it directly it should be in reqs. I missed that you were using it in eval_with_uncertainties